addons/sale/tests/test_sale_report.py

```diff
@@ -43,9 +43,9 @@ def test_sale_report_foreign_currency(self):
 
         # Create corresponding pricelists and rates.
         pricelists = self.env['product.pricelist'].create([
-            {'name': 'Pricelist (USD)', 'currency_id': usd.id},
-            {'name': 'Pricelist (EUR)', 'currency_id': eur.id},
-            {'name': 'Pricelist (ARS)', 'currency_id': ars.id},
+            {'name': 'Pricelist (USD)', 'currency_id': usd.id, 'company_id': self.env['res.company']},
+            {'name': 'Pricelist (EUR)', 'currency_id': eur.id, 'company_id': self.env['res.company']},
+            {'name': 'Pricelist (ARS)', 'currency_id': ars.id, 'company_id': self.env['res.company']},
```

Comment:
Quite the opposite in fact, in the follow-up of this test, we are then using these pricelists on 2 different companies... As the default is the env company, you should instead voluntarily enforce that this value stays empty. It's strange that the SO can be made with a pricelist belonging to another company, in fact. I don't see why this test isn't currently failing... :thinking:

Reply:
My mistake, it's an empty recordset, I wrongly read `cls.env.company`